tests/unit/test_read_configs_include.py

```diff
@@ -547,6 +567,156 @@ def test_allowed_no_override_old_variables_when_set(self, read_config_include_fi
         print(proc.stdout)
         assert split_and_strip(get_command_stdout(proc))[-1] == "pavics.example.com"
 
+    @pytest.mark.parametrize("from_name,to_name",
+        (("ENABLE_JUPYTERHUB_MULTI_NOTEBOOKS", "JUPYTERHUB_ENABLE_MULTI_NOTEBOOKS"),
+         ("JUPYTERHUB_ENABLE_MULTI_NOTEBOOKS", "ENABLE_JUPYTERHUB_MULTI_NOTEBOOKS")))
+    def test_formatting_preserved_from_old_to_new_var_vice_versa(self, read_config_include_file,
+                                                                 exit_on_error, from_name, to_name):
+        """
+        Test that formatting (new lines, leading spaces, quotes) are preserved when old var
+        value is transfered to new var and vice-versa.  This is important during template expansion.
+        """
+        expected = ("\n"
+                    "    # python code requires keeping formatting  \n"
+                    "    {'user': 'pass'}\n")
+
+        extra = {
+            from_name: f"\"{expected}\"",
+            "BIRDHOUSE_BACKWARD_COMPATIBLE_ALLOWED": "True",
+        }
+        proc = self.run_func(
+            read_config_include_file,
+            extra,
+            f'echo "${to_name}"',
+            exit_on_error=exit_on_error,
+        )
+        print(proc.stdout)
+        assert "\n".join(get_command_stdout(proc).split("\n")[-4:]) == expected
+
+    def test_template_expansion_enabled_for_old_var(self, read_config_include_file, exit_on_error):
```

Comment:
This should be split into multiple test functions. You've got multiple calls to proc.run_func and multiple asserts for each. 

For organizational purposes you could even move this to its own class and make some of the repeated code into fixtures.

Otherwise, it is unclear what this function is testing because it is testing too many things at once.

Reply:
Yeah let me try to clean this up.